play_book_pdf_tool/play_book_pdf_tool.py

```diff
@@ -90,14 +90,12 @@ def add_metadata(base_path, pdf):
     title, authors, pub_date, num_pages, publisher = itemgetter('title', 'authors', 'pub_date', 'num_pages',
                                                                 'publisher')(manifest['metadata'])
 
-    year, month, day = [int(x) for x in pub_date.split('.')]
-
     with pdf.open_metadata() as pdf_metadata:
         pdf_metadata['dc:title'] = title
         pdf_metadata['dc:creator'] = [author.strip() for author in authors.split(',')]
         pdf_metadata['dc:language'] = [language]
         pdf_metadata['dc:publisher'] = [publisher]
-        pdf_metadata['xmp:CreateDate'] = datetime.datetime(year, month, day).isoformat()
+        pdf_metadata['xmp:CreateDate'] = pub_date.replace('.', '-')
```

Comment:
@coderabbitai Add validation to verify that it's a valid `xmp:CreateDate`. Log a warning if it's not valid, ignore this piece of metadata, and continue the program.

Reply:
Shorten the code and only show what you added/deleted.

Reply:
@coderabbitai Validate that the date is in the following format:

```
YYYY
YYYY-MM
YYYY-MM-DD
YYYY-MM-DDThh:mmTZD
YYYY-MM-DDThh:mm:ssTZD
YYYY-MM-DDThh:mm:ss.sTZD
```

Reply:
I don't like that ugly RegEx. Rewrite it so that each possible format is verified independently.